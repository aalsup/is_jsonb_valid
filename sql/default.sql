-- invalid type for default
-- valid when property is specified
SELECT is_jsonb_valid('{"properties":{"foo":{"type":"integer","default":[]}}}', '{"foo":13}');
-- still valid when the invalid default is used
SELECT is_jsonb_valid('{"properties":{"foo":{"type":"integer","default":[]}}}', '{}');
-- invalid string value for default
-- valid when property is specified
SELECT is_jsonb_valid('{"properties":{"bar":{"type":"string","minLength":4,"default":"bad"}}}', '{"bar":"good"}');
-- still valid when the invalid default is used
SELECT is_jsonb_valid('{"properties":{"bar":{"type":"string","minLength":4,"default":"bad"}}}', '{}');