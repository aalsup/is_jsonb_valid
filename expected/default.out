-- invalid type for default
-- valid when property is specified
SELECT is_jsonb_valid('{"properties":{"foo":{"type":"integer","default":[]}}}', '{"foo":13}');
 is_jsonb_valid 
----------------
 t
(1 row)

-- still valid when the invalid default is used
SELECT is_jsonb_valid('{"properties":{"foo":{"type":"integer","default":[]}}}', '{}');
 is_jsonb_valid 
----------------
 t
(1 row)

-- invalid string value for default
-- valid when property is specified
SELECT is_jsonb_valid('{"properties":{"bar":{"type":"string","minLength":4,"default":"bad"}}}', '{"bar":"good"}');
 is_jsonb_valid 
----------------
 t
(1 row)

-- still valid when the invalid default is used
SELECT is_jsonb_valid('{"properties":{"bar":{"type":"string","minLength":4,"default":"bad"}}}', '{}');
 is_jsonb_valid 
----------------
 t
(1 row)

